File: benchmarks/perl_php/oop.pl
package Counter;
sub new { return bless({count=>0}, $_[0]); }
sub increment { $_[0]->{count}++; }
sub get { return $_[0]->{count}; }
package main;
my $it=5000000; my $c=Counter::new("Counter");
for (my $i=0;$i<$it;$i++){ $c->increment(); }
print "oop: $it method calls, count=".$c->get()."\n";

File: benchmarks/perl_php/js/oop.js
class Counter { constructor(){this.count=0;} increment(){this.count++;} get(){return this.count;} }
const it=5000000; const c=new Counter();
for(let i=0;i<it;i++){ c.increment(); }
console.log("oop: "+it+" method calls, count="+c.get());
